### PR TITLE
Authentication with devise gem

### PR DESCRIPTION
**Pull request by Atifzada**
### Following changes have been made during this PR:

- [x] Devise setup
- [x] Sign up a new user
- [x] Post as a user
- [x] Fix linter errors

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
```
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb]: - [x] [OPTIONAL]
Generally, if a user not signed in, the user should be redirected to the login page, however, when enter the app it always assumes that the user is signed in. Let's endeavor to improve upon this aspect.

**For the first time app assumes the user is already signed in:**
image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+<%= render "devise/shared/links" %>
```
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb]: - [x] Nice job making the signup/login page work 👌 However, this activity requires the modification of the views in _devise_ for the registration and login to match the styling of the [provided wireframe(s)](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md). Kindly ensure that the form (inputs and buttons) stretches across the screen and is centered accordingly for both **Login** and **Signup** pages, as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md):

**Your Signup view:**        |  **Your login view:**        |  **Wireframe login view (Required):**
:-------------------------:|:-------------------------:|:-------------------------:
Home Page | Home Page | Home Page

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Atif, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Authentication is coming up well so far, good job!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
Kindly include appropriate instructions in your ReadMe under the _Getting Started_ section, on how to run/set up your project.

Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (Atifzada) [reply app/views/devise/sessions/new.html.erb]: Thank you for the helpful review. surely I will work on it

**Review by algerina — APPROVED** [review]
Hi @Atifzada  ,

Great job so far!  

May I highlight the following: 
- [x] Your  have a descriptive PR and commit history👏  ✔️ 
- [x] Your implemented all the requirements of the previous TSE✔️ 

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**